sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java

```diff
@@ -216,7 +220,11 @@ void loadBalance() {
                 logger.info("This event processor owns {} partitions and shouldn't own more",
                     ownerPartitionMap.get(ownerId).size());
                 // renew ownership of already owned partitions
-                checkpointStore.claimOwnership(ownerPartitionMap.get(this.ownerId)).subscribe();
+                checkpointStore.claimOwnership(partitionPumpManager.getPartitionPumps().keySet()
+                    .stream()
+                    .map(partitionId -> createPartitionOwnershipRequest(partitionOwnershipMap, partitionId))
+                    .collect(Collectors.toList()))
```

Comment:
Curious why we are collecting these into a list? it only emits when the flux is complete.

Reply:
`checkpointStore.claimOwnership()` method takes a list of partition ownership instances.